framework/src/materials/MaterialPropertyInterface.C

```diff
@@ -192,6 +192,30 @@ MaterialPropertyInterface::getMaterialByName(const std::string & name, bool no_w
   return *discrete;
 }
 
+std::set<MaterialBase *>
+MaterialPropertyInterface::getSupplyerMaterials()
```

Comment:
Do you want just the materials that supply your dependencies or will you also need the materials that supply the properties satisfying your dependency materials dependencies? If the latter, then there is already a method for that: `MaterialBase::buildRequiredMaterials`

Reply:
Good to know, but that's in MaterialBase and I need it in UserObject and AuxKernel. 

You are right though, that I might need it for the whole dependency chain, as technically I suppose, the stateful material property initialization could rely on initialized stateful properties of other materials... (?)

Reply:
Oh, it's static. I'll try using that function instead!

Reply:
> the stateful material property initialization could rely on initialized stateful properties of other materials... (?)

I don't know this off the top of my head